.github/workflows/e2e-backend-switching-reuseable.yml

```diff
@@ -0,0 +1,192 @@
+name: Backend-switching-tests-gardener (reusable)
+
+on:
+  workflow_call:
+    inputs:
+      eventing-manager-image:
+        description: Container image of eventing-manger which needs to be tested.
+        required: true
+        type: string
+      kube-version:
+        description: Kubernetes version for the Gardener cluster.
+        required: true
+        type: string
+      cluster-name-prefix:
+        description: Prefix to use as prefix for name of Gardener cluster.
+        required: false
+        type: string
+        default: "ghem-"
+
+jobs:
+  backend-switching:
+    runs-on: ubuntu-latest
+    env:
+      KYMA_STABILITY: "unstable"
+      KYMA: "./hack/kyma"
+      MANAGER_IMAGE: ${{ inputs.eventing-manager-image }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Cache Binaries
+        id: cache-binaries
+        uses: actions/cache@v3
+        with:
+          path: bin
+          key: ${{ runner.os }}-bin
+
+      - name: Install Kyma CLI
+        run: |
+          make kyma
+
+      - name: Provision Gardener cluster
+        env:
+          GARDENER_CLUSTER_VERSION: ${{ inputs.kube-version }}
+          CLUSTER_PREFIX: ${{ inputs.cluster-name-prefix }}
+          GARDENER_REGION: "eu-west-1"
+          GARDENER_ZONES: "eu-west-1a"
+          GARDENER_PROJECT_NAME: ${{ vars.GARDENER_PROJECT_NAME }}
+          GARDENER_PROVIDER_SECRET_NAME: ${{ vars.GARDENER_PROVIDER_SECRET_NAME_AWS }}
+          MACHINE_TYPE: "c4.xlarge"
+          SCALER_MIN: "1"
+          SCALER_MAX: "2"
+          RETRY_ATTEMPTS: "2"
+          GARDENER_KYMATUNAS: ${{ secrets.GARDENER_KYMATUNAS }}
+        run: |
+          : ### setup Gardener kubeconfig.
```

Comment:
what does the `:` do?

Reply:
Its a hacky way to add comments in the `run` section. Otherwise it does not work if we start only with `#`.